source/Nuke.Common/Tools/OctoVersion/OctoVersionAttribute.cs

```diff
@@ -22,27 +24,232 @@ namespace Nuke.Common.Tools.OctoVersion
+        private int? _major;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Major version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string MajorParameter { get; set; }
+
+        /// <summary>
+        /// The Minor version number to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public int Minor { get { return _minor.GetValueOrDefault(); } set { _minor = value; } }
+        private int? _minor;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Minor version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string MinorParameter { get; set; }
+
+        /// <summary>
+        /// The Patch version number to pass to OctoVersion
+        /// </summary>
+        [CanBeNull] public int Patch { get { return _patch.GetValueOrDefault(); } set { _patch = value; } }
+        private int? _patch;
+
+        /// <summary>
+        /// The name of a Nuke parameter containing the Patch version number to be passed to OctoVersion
+        /// </summary>
+        [CanBeNull] public string PatchParameter { get; set; }
 
         public override object GetValue(MemberInfo member, object instance)
         {
+            // Wrap log output in a NUKE Section
+            using (Logger.Block("[OctoVersion] Injection"))
+            {
+                LogAttributeProperties();
+
+                return GetValueInternal(member, instance);
+            }
+        }
+
+        private void LogAttributeProperties()
+        {
+            Logger.Trace("Defined Properties");
+            Logger.Trace($"  Framework:                 {Framework ?? "null"}");
+            Logger.Trace($"  UpdateBuildNumber:         {UpdateBuildNumber}");
+            Logger.Trace($"  AutoDetectBranch:          {_autoDetectBranch?.ToString() ?? "null"}");
+            Logger.Trace($"  Branch:                    {Branch ?? "null"}");
+            Logger.Trace($"  Major:                     {_major?.ToString() ?? "null"}");
+            Logger.Trace($"  Minor:                     {_minor?.ToString() ?? "null"}");
+            Logger.Trace($"  Patch:                     {_patch?.ToString() ?? "null"}");
+            Logger.Trace($"  AutoDetectBranchParameter: {AutoDetectBranchParameter ?? "null"}");
+            Logger.Trace($"  BranchParameter:           {BranchParameter ?? "null"}");
+            Logger.Trace($"  MajorParameter:            {MajorParameter ?? "null"}");
+            Logger.Trace($"  MinorParameter:            {MinorParameter ?? "null"}");
+            Logger.Trace($"  PatchParameter:            {PatchParameter ?? "null"}");
+        }
+
+        private object GetValueInternal(MemberInfo member, object instance)
+        {
+            // Get AutoDetectBranch / AutoDetectBranchParameter
+            bool? octoVersionAutoDetectBranch = null;
```

Comment:
These blocks here are quite repetitive. Also, I would consider the following:

For instance with `AutoDetectBranch`, a user might want to set a certain value, but we could as well allow the parameter `AutoDetectBranchParameter` to override it. So the logic is essentially `GetValue(AutoDetectBranchParameter) ?? AutoDetectBranch`, which I think is much more flexible.

Reply:
Yes they are repetitive 🤣 . For the intial review i think it was a good idea to keep it all spelled out so we could discuss.
I like your idea of the override behaviour - not only does it make the code clearer it also removes alot of the ControlFlow assertions that were trying to make the pairs of properties mutually exclusive.  I will look to update it

Reply:
implemented as per your suggestion :+1: